kaldb/src/main/java/com/slack/kaldb/logstore/search/KaldbDistributedQueryService.java

```diff
@@ -151,6 +154,21 @@ private void updateStubs() {
     try {
       List<SearchResult<LogMessage>> searchResults =
           searchFuture.get(READ_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+      // we want to be super sure the inner LOG.debug statements are not getting evaluated and then
+      // thrown away because debug is enabled in production as this is a hot code path
+      if (LOG.isDebugEnabled()) {
```

Comment:
I would just do this as a normal `LOG.debug` instead of adding a special check - additionally, when constructing the value we should be using `{}` instead of constructing the string with concatenation, as it defers the computation.

Reply:
I'm still thinking of keeping this because we will save on iterating the loop of hundreds of servers potentially

Reply:
I changed the comment to reflect the new reason